core/src/main/scala/quasar/destination/avalanche/AvalancheDestination.scala

```diff
@@ -218,6 +227,18 @@ final class AvalancheDestination[F[_]: ConcurrentEffect: ContextShift: MonadReso
     fr"DROP TABLE IF EXISTS" ++ Fragment.const(table)
   }
 
+  private def truncateTableQuery(tableName: TableName): Fragment = {
+    val table = tableName.name
+
+    fr"MODIFY" ++ Fragment.const(table) ++ fr"TO TRUNCATED"
+  }
+
+  private def existanceTableQuery(tableName: TableName): Fragment = {
+    val table = tableName.name.toLowerCase.substring(1, tableName.name.length()-1)
```

Comment:
Ah, so we're escaping the table name too eagerly. This works, but I think a better solution would be to move escaping to where it is needed, rather than when extracting the table name from the `ResourcePath`. Then we wouldn't have any assumptions about `TableName` representing an escaped identifier and the value could be used here without modification.

Reply:
We've decided to handle this change in a separate PR.